r/R/dplyr-group-by.R

```diff
@@ -18,16 +18,33 @@
 
 # The following S3 methods are registered on load if dplyr is present
 
+#' @importFrom rlang quo_name
```

Comment:
Just checking for my own understanding - we have some function imported specified in arrow-package.R, and others next to the function that calls them - what determines this?

Reply:
My read on how we do it in the Arrow package is that we import if:

* it's in the package (i.e. under the R/ directory, as opposed to in tests)
* the dependency is _not_ a `SUGGEST`

otherwise we (tend to) `pkg::` (e.g. a bunch of `rlang::eval_tidy` in `helper-expectation.R`.  I'm sure there are exceptions to this and looking for `::` in `R/` I see a few just scanning.

I've seen others recommend that using a function a small number of times it's better to `pkg::` that and once some threshold is reached then switch to importing. Or use `pkg::` for the first call in a file / package to make it clear that's what one is referring to.

Reply:
I think this is about the `@importFrom`, which Nic is right, we have most of them grouped in one place. I'm indifferent about how we do it but probably makes sense to follow convention and for these rlang ones at least, which tend to get used in multiple places, keep them grouped in arrow-package.R

Reply:
OOOH, I see, I misread it at first. Yeah I can move it over to arrow-package.R

Reply:
Yeah, I could have been more clear in how I phrased that, thanks for clarifying there Neal!

Reply:
To your points Jon, IMO any function from a package we Import should be in `importsFrom` and be used without `::`, and we only use `::` for Suggested packages.